### DISCUSSION TITLE
Problem with @script attribute being added repeatedly in nesting components

### DISCUSSION THREAD

**Discussion by smertozturk** [livewire/livewire#7366] [7. Bugs (archived)]
### Livewire version

v3.2.1

### Which PHP version are you using?

PHP 8.2

### Steps To Reproduce

The javascript codes contained in @script - @endscript in a child component are added to the page every time this component is called again. For example, in the code below, I run toastr as a result of a dispatch. When this child component is rendered for the second time, the same script is added to the page for the second time.

My Nested Component Parent Area;
```
<a href="javascript:void(0)" wire:click="next('profile.account')"">Account Details</a>
<a href="javascript:void(0)" wire:click="next('profile.notification')"">Account Notification</a>
<livewire:dynamic-component :is="$current" :key="Str::random()"/>
```

Parent Component Livewire file (next function);

```
public function next($component)
    {
        $this->current = $component;
    }
```

Child component;
```
<div>
    //....
</div>
@script
<script>
    Livewire.on('account-updated', function (e) {
        toastr.success(e.title);
    });
</script>
@endscript
```

I'm not sure if this is a bug, sorry if that's what it's supposed to be.

### Please confirm (incomplete submissions will not be addressed)

- [x] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by WilliamPinto-Olmos** [comment]
hey @sameter123 try wrapping the @script directive in the @once directive.

Calling the next component twice will log 'hey' twice to the console.

```html
<div>
    ...
    @script
    <script>
        console.log('hey')
    </script>
    @endscript
</div>
```

Calling the next component twice (or multiple times) will log 'hey' once to the console.

```html
<div>
    ...

    @once
      @script
        <script>
            console.log('hey')
        </script>
      @endscript
    @endonce
</div>
```

**Reply by smertozturk** [reply]
Oh, i tried with your suggest and i noticed this;

Inside of my script, i have livewire.on and i think this register event listener for this event first time, still when calling next component twice, it will register again and account-updated event work two time.
```
Livewire.on('account-updated', function (e) {
                toastr.success(e.title);
            });
```

Any suggest for one time register, i don't want to move this script to parent component because this child component need own script for re-usable.

**Reply by WilliamPinto-Olmos** [reply]
Im currently testing this on version 3.2.1

Given the next child component view.

```html
<div>
    <button x-data @click="Livewire.dispatch('some-event')">
        Dispatch some event from child
    </button>

    @once
        @script
            <script>
                Livewire.on('some-event', () => console.log("Hello from component"));
            </script>
        @endscript
    @endonce
</div>
```

And the next parent component view

```html
<div>
   <livewire:some-namespace.some-component />    
   <livewire:some-namespace.some-component />    

   <button x-data @click="Livewire.dispatch('some-event')">
      Dispatch some event from parent
   </button>
</div>
```
The `some-event` event is only called once per button clicked (any of them). So you might wanna check your livewire version, otherwise i think it could be the fact that you are removing programatically the component, but if thats the case I wouldn't know how to fix it, sorry:c
